lib/Common/CommonPal.h

```diff
@@ -611,19 +611,20 @@ HRESULT ULongMult(ULONG ulMultiplicand, ULONG ulMultiplier, ULONG* pulResult);
 template <class TryFunc, class FinallyFunc>
 void TryFinally(const TryFunc& tryFunc, const FinallyFunc& finallyFunc)
 {
-    bool hasException = true;
-    try
+    class FinallyObject
```

Comment:
Catch is called before the stack is unwound.  Rethrow costs a lot of stack space.  WIth the Dtors, you don't consume extra stack space when you have many of these TryFinally nested.

---
In reply to: [100933897](https://github.com/Microsoft/ChakraCore/pull/2519#discussion_r100933897) [](ancestors = 100933897)